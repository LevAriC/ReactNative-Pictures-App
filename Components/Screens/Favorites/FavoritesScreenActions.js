import { ADD_TO_FAVORITES } from './FavoritesScreenActionTypes'

const addToFavorites = (pic) => ({
    type: ADD_TO_FAVORITES,
    data: { pic }
})

const handleAddToFavorites = (pic) => async dispatch => {
    console.log('handleAddToFavorites')
    dispatch(addToFavorites(pic))
}

export default {
    handleAddToFavorites
}