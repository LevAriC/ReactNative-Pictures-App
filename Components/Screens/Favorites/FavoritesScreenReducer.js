import { ADD_TO_FAVORITES } from './FavoritesScreenActionTypes'

const initialState = {
    favoritesList: [],
}

export default (state = initialState, action) => {
    switch (action.type) {
        case ADD_TO_FAVORITES:
            return {
                ...state,
                favoritesList: [...state.favoritesList, action.data.pic]
            }
        default:
            return state
    }
}