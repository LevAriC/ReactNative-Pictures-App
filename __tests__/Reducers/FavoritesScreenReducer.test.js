import { applyMiddleware, createStore, compose } from 'redux'
import thunk from 'redux-thunk'
import rootReducer from '../../Components/rootReducer'
import FavoritesScreenActions from '../../Components/Screens/Favorites/FavoritesScreenActions'
import FavoritesScreenReducer from '../../Components/Screens/Favorites/FavoritesScreenReducer'

const middleware = applyMiddleware(thunk)
const composedEnhancers = compose(middleware)
const initialState = {}

it('FavoritesScreen - Initial State', () => {
    const store = createStore(rootReducer, initialState, composedEnhancers)
    expect(store.getState().FavoritesScreen.favoritesList).toMatchSnapshot()
    expect(store.getState().FavoritesScreen.view).toMatchSnapshot()
    expect(store.getState().FavoritesScreen.isLoading).toMatchSnapshot()
    expect(store.getState().FavoritesScreen.bigPic).toMatchSnapshot()
});

it('FavoritesScreen - Adding To Favs', () => {
    const store = createStore(rootReducer, initialState, composedEnhancers)
    store.dispatch(FavoritesScreenActions.addToFavorites({pic: 'surf'}))
    expect(store.getState().FavoritesScreen.favoritesList).toHaveLength(1)
    expect(store.getState().FavoritesScreen.favoritesList[0]).toEqual({pic: 'surf'})
    store.dispatch(FavoritesScreenActions.saveBigPic({pic: 'surf'}))
    expect(store.getState().FavoritesScreen.bigPic).toEqual({pic: 'surf'})
});

it('HomeScreen - Change View', () => {
    const store = createStore(rootReducer, initialState, composedEnhancers)
    expect(store.getState().FavoritesScreen.view).toMatchSnapshot()
    store.dispatch(FavoritesScreenActions.handleChangeView("List"))
    expect(store.getState().FavoritesScreen.view).toMatchSnapshot()
    store.dispatch(FavoritesScreenActions.handleChangeView("Picture"))
    expect(store.getState().FavoritesScreen.view).toMatchSnapshot()    
});
