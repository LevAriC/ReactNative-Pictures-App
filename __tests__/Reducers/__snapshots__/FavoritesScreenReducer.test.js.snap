// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesScreen - Initial State 1`] = `Array []`;

exports[`FavoritesScreen - Initial State 2`] = `"Grid"`;

exports[`FavoritesScreen - Initial State 3`] = `true`;

exports[`FavoritesScreen - Initial State 4`] = `Object {}`;

exports[`HomeScreen - Change View 1`] = `"Grid"`;

exports[`HomeScreen - Change View 2`] = `"List"`;

exports[`HomeScreen - Change View 3`] = `"Picture"`;
